src/faction_camp.cpp

```diff
@@ -1420,20 +1408,136 @@ void basecamp::craft_construction( npc &p, const std::string &cur_id, const std:
+        //Needs to be plowed to match json
+        if( is_dirtmound( pos, bay_json, bay ) && is_unplowed( pos, bay ) ) {
+            plots_plow += 1;
+        }
+        if( is_dirtmound( pos, bay, bay ) ) {
+            plots_empty++;
+        }
+        if( bay.furn( pos ) == furn_str_id( "f_plant_harvest" ) && !bay.i_at( pos ).empty() ) {
+            const item &seed = bay.i_at( pos ).front();
+            if( is_valid_seed( seed ) ) {
+                harvestable++;
+                const islot_seed &seed_data = *seed.type->seed;
+                item tmp = item( seed_data.fruit_id, calendar::turn );
+                if( std::find( plant_names.begin(), plant_names.end(),
+                               tmp.type_name( 3 ) ) != plant_names.end() ) {
+                    plant_names.push_back( tmp.type_name( 3 ) );
+                }
+            }
         }
     }
 
+    int total_c = 0;
+    for( const std::string &i : plant_names ) {
+        if( total_c < 5 ) {
+            crops += "\t" + i + " \n";
+            total_c++;
+        } else if( total_c == 5 ) {
+            crops += "+ more \n";
+            break;
+        }
+    }
+
+    if( plow ) {
+        return plots_plow;
+    } else if( plant ) {
+        return plots_empty;
+    } else if( harvest ) {
+        return harvestable;
+    }
+    return 0;
+}
+
+void talk_function::camp_farm_start( npc &p, const std::string &dir, const tripoint &omt_tgt,
+                                     farm_ops op )
+{
+    bool plow = op & farm_ops::plow;
+    bool plant = op & farm_ops::plant;
+    bool harvest = op & farm_ops::harvest;
+    std::string crops_null;
+    size_t plots_cnt = camp_farm_act( omt_tgt, crops_null, op );
+    time_duration work = 0_minutes;
+    npc_ptr comp = nullptr;
+    if( harvest && plots_cnt ) {
+        work += 3_minutes * plots_cnt;
+        comp = individual_mission( p, _( "begins to harvest the field..." ),
+                                   "_faction_exp_harvest_" + dir, false, {}, "survival" );
+    }
+    if( plant && plots_cnt ) {
+        inventory total_inv = g->u.crafting_inventory();
+        std::vector<item *> seed_inv = total_inv.items_with( []( const item & itm ) {
+            return itm.is_seed() && itm.typeId() != "marloss_seed" && itm.typeId() != "fungal_seeds";
+        } );
+        if( seed_inv.empty() ) {
+            popup( _( "You have no additional seeds to give your companions..." ) );
+            return;
+        }
+        std::vector<item *> plant_these;
+        plant_these = individual_mission_give_equipment( seed_inv,
+                      _( "Which seeds do you wish to have planted?" ) );
+        size_t seed_cnt = 0;
+        for( item *seeds : plant_these ) {
+            seed_cnt += seeds->count();
+        }
+        size_t plots_seeded = std::min( seed_cnt, plots_cnt );
+        work += 1_minutes * plots_seeded;
+        comp = individual_mission( p, _( "begins planting the field..." ),
```

Comment:
This assigns `comp` again (and later again in the `if` branch below, but the final to adjust `comp->companion_mission_time` is only done for the last assigned companion (or none if `individual_mission` yields `null`). This seems wrong. It should be applied to all npcs or none of them.